contracts/test-contracts/MockContract.sol

```diff
@@ -0,0 +1,66 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity 0.8.20;
+
+contract MockContract {
+    event Called(uint256 value, bytes data);
+
+    struct CallResult {
+        bytes calldata_;
+        bool failure;
+        bytes returnData;
+    }
+
+    CallResult[] private results;
```

Comment:
You can use [EnumerableSet](https://github.com/OpenZeppelin/openzeppelin-contracts/blob/master/contracts/utils/structs/EnumerableSet.sol#L41), so you don't need to traverse all the array when you add something. Maybe not a concern for the current tests but would be cheaper in general and in the future better test coverage this may create a problem anyway.

Reply:
Actually, I think we need mapping here. I mean, of course, we can change this set to work as mapping and change inner types. But do you think it is worth it?